### DISCUSSION TITLE
Unify/relate Foldable and Iterable

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#9335] [Q&A]
Or help me understand why they are different

**Comment by magnus-madsen** [comment]
Foldable has foldRight. Foldable is a "pure functional abstraction".

**Reply by JonathanStarup** [reply]
Iterable in our representation is not IterableOnce. Why can you not define foldRight with an iterator?

**Reply by JonathanStarup** [reply]
Maybe to clarify - they seem to have a close connection that we do not represent in code:

Clearly you can iterate everything that is Foldable (just use foldLeft to create an Iterator). This implies that everything Iteratable is Foldable.

It seems to me that you can implement Foldable if you have an Iterator instance which implies that everything Foldable is Iterable.

This seeming implication/equality creates confusion when you just "want a collection of ordered things" - should you require an Iterable or a Foldable

**Comment by JonathanStarup** [comment]
This highlights the practical problem

![image](https://github.com/user-attachments/assets/ce8d7981-92a2-4a55-b409-b6d6c7ff3a77)
